astropy/uncertainty/tests/test_containers.py

```diff
@@ -0,0 +1,72 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Test that Distribution works with classes other than ndarray and Quantity."""
+
+import numpy as np
+import pytest
+from numpy.testing import assert_array_equal
+
+import astropy.units as u
+from astropy.coordinates import Angle, Latitude, Longitude
+from astropy.uncertainty import Distribution
+
+
+class TestAngles:
+    @classmethod
+    def setup_class(cls):
+        cls.a = np.arange(27.0).reshape(3, 9)
+        cls.d = Distribution(cls.a)
+        cls.q = cls.a << u.deg
+        cls.dq = Distribution(cls.q)
+
+    @pytest.mark.parametrize("angle_cls", [Angle, Longitude, Latitude])
```

Comment:
maybe should add `EarthLocation` to this list? Although that might be a whole can of worms.

Reply:
`EarthLocation` will be interesting, since it is already a structured type. That will really need some thought, since presumably each sample then has `x,y,z` inside, but we still have to be sure `el_dist['x']` works... In other words, I'm pretty sure this will not yet work!